code/modules/projectiles/guns/energy/pulse.dm

```diff
@@ -5,6 +5,7 @@
 	item_state = null
 	w_class = WEIGHT_CLASS_BULKY
 	force = 10
+	modifystate = 1

```

Comment:
In code it states its either 0 or 1, this is how I figured out why the Pulse weapon sprites were not working when it came to swapping fire modes. modifystate = 1 is also how it works for energy_guns.